gui/src/controllermanager.cpp

```diff
@@ -56,7 +57,15 @@ void ControllerManager::UpdateAvailableControllers()
 	{
 		if(!SDL_IsGameController(i))
 			continue;
-		current_controllers.insert(SDL_JoystickGetDeviceInstanceID(i));
+
+		// We'll try to identify pads with Motion Control
+		SDL_JoystickGUID guid = SDL_JoystickGetGUID(SDL_JoystickOpen(i));
+		char guid_str[256];
+		SDL_JoystickGetGUIDString(guid, guid_str, 256);
+
+		int nButtons = SDL_JoystickNumButtons(SDL_JoystickOpen(i));
+		if(!(nButtons == 0 && chiaki_motion_controller_guid_list.indexOf(guid_str) > -1))
+			current_controllers.insert(SDL_JoystickGetDeviceInstanceID(i));
```

Comment:
Joysticks opened with `SDL_JoystickOpen()` must be closed and early return should be used.
So better to restructure this to something like this (untested, but I think you get the idea):

```c++
		if(chiaki_motion_controller_guids.contains(guid_str))
		{
			SDL_Joystick *joy = SDL_JoystickOpen(i);
			if(joy)
			{
				bool no_buttons = SDL_JoystickNumButtons(joy) == 0;
				SDL_JoystickClose(joy);
				if(no_buttons)
					continue;
			}
		}

		current_controllers.insert(SDL_JoystickGetDeviceInstanceID(i));
```